hcl2template/parser.go

```diff
@@ -94,11 +94,21 @@ func (p *Parser) parse(filename string, vars map[string]string) (*PackerConfig,
 	// can use input variables so we decode them firsthand.
 	{
 		for _, file := range files {
-			diags = append(diags, p.decodeInputVariables(file, cfg)...)
+			diags = append(diags, cfg.decodeInputVariables(file)...)
 		}
+
+		var locals []*Local
 		for _, file := range files {
-			diags = append(diags, p.decodeLocalVariables(file, cfg)...)
+			moreLocals, morediags := cfg.parseLocalVariables(file)
+			diags = append(diags, morediags...)
+
+			if locals == nil {
+				locals = moreLocals
+			} else {
+				locals = append(locals, locals...)
+			}
```

Comment:
I think this is not necessary, we can just `locals = append(locals, locals...)` here